scripts/generate-sitemap.js

```diff
@@ -13,63 +13,147 @@ async function readSchemaJson(filePath) {
   }
 }
 
-async function processDirectory(dirPath, relativePath = "") {
-  const entries = await fs.readdir(dirPath, { withFileTypes: true });
-  let paths = [];
-
-  for (const entry of entries) {
-    const entryPath = path.join(dirPath, entry.name);
-
-    if (entry.name === "index.json") {
-      continue;
+async function processFile(fullPath, relativePath, name) {
+  const schemaPath = _path.join(fullPath, name);
+  const schemaData = await readSchemaJson(schemaPath);
+  if (!schemaData) {
+    return null;
+  }
+  const pageName = name.split(".")[0];
+  const permalink = _path.join(relativePath, pageName);
+  const siteMapEntry = {
+    permalink,
+    title:
+      schemaData.page.title ||
+      pageName.charAt(0).toUpperCase() + pageName.slice(1),
+  };
+  // check if file is actually an index page for a directory
+  try {
+    const directoryPath = _path.join(fullPath, name.split(".")[0]);
+    const stats = await fs.stat(directoryPath);
+    if (stats.isDirectory()) {
+      return {
+        ...siteMapEntry,
+        children: await processDirectory(directoryPath, permalink),
+      };
     }
+    return siteMapEntry;
+  } catch (error) {
+    return siteMapEntry;
+  }
+}
 
-    if (entry.isDirectory()) {
-      const subRelativePath = path.join(relativePath, entry.name);
+// Returns new index files that were generated as they might have to be added to _pages.json
+async function generateIndexFilesForDirectoriesIfMissing(dirEntries) {
+  const newIndexFiles = [];
+  for (const dirEntry of dirEntries) {
+    const indexFilePath = _path.join(dirEntry.path, dirEntry.name + ".json");
+    try {
+      await fs.access(indexFilePath);
+    } catch (error) {
+      const dirName = dirEntry.name.split(".")[0].replace(/-/g, " ");
+      await fs.writeFile(
+        indexFilePath,
+        JSON.stringify(
+          {
+            version: "0.1.0",
+            layout: "content",
+            page: {
+              title: dirName.charAt(0).toUpperCase() + dirName.slice(1),
+              contentPageHeader: {},
```

Comment:
nit: might need to add a sane default for the `summary` prop here, something like "Pages in ${title}" or something.